### DISCUSSION TITLE
Widget did not get activated even when the tab is the current one

### DISCUSSION THREAD

**Discussion by bforget163** [eclipse-theia/theia#8417] [General] [answered]
Hi,
I am adding a Widget to theia extension, my widget implements SaveableSource interface as well as include a Saveable object. However the `Save` command from menu does not work when the content changed and the widget got `dirty` status. The saveable event works fine with the `Save All` command, so I think this should be the issue to widget's activate?

I can see the below warning in Chrome console, also tried the [example-4 ](https://github.com/akosyakov/theia-training/tree/solution-4)from @akosyakov and got the same warning.

`Widget was activated, but did not accept focus after 2000ms`

![image](https://user-images.githubusercontent.com/33746058/90997299-c4cffc80-e5f3-11ea-887e-ad709bfb6b98.png)

**Comment by akosyakov** [comment]
A widget should implement `Widget.onActivateRequest` to get rid of this warning. Activation means to focus something within the widget. Theia does not know what is something and delegates to a widget.

**Comment by kittaakos** [comment, marked as answer]
> also tried the [example-4 ](https://github.com/akosyakov/theia-training/tree/solution-4)from @akosyakov and got the same warning.

As suggested above, you have to handle the activation manually. The following will fix the widget activation in the example:
```diff
diff --git a/theia-training/src/browser/jsonschema-form-widget.tsx b/theia-training/src/browser/jsonschema-form-widget.tsx
index 74b7ca1..2e15614 100644
--- a/theia-training/src/browser/jsonschema-form-widget.tsx
+++ b/theia-training/src/browser/jsonschema-form-widget.tsx
@@ -56,5 +56,11 @@ export class JsonschemaFormWidget extends BaseWidget {
         ReactDOM.render(model ? <JsonschemaFormView model={model} modelService={this.modelService} /> : null!, this.viewNode);
     }
 
+    protected onActivateRequest(message: Message): void {
+        super.onActivateRequest(message);
+        // Set the focus on the desired HTML element.
+        this.viewNode.focus();
+    }
+
 }
```

**Reply by bforget163** [reply]
Thanks @kittaakos and @akosyakov , I tried the solution above and did realized that my widget should at least have one focusable HTML element (e.g. Input) to get real focus -- I guess that's how theia check which is the current activated widget.
